tsl/src/compression/compress_utils.c

```diff
@@ -436,6 +437,80 @@ decompress_remote_chunk(FunctionCallInfo fcinfo, const Chunk *chunk, bool if_com
 	return success;
 }
 
+/*
+ * Create a new compressed chunk using existing table with compressed data.
+ *
+ * chunk_relid - non-compressed chunk relid
+ * chunk_table - table containing compressed data
+ */
+Datum
+tsl_create_compressed_chunk(PG_FUNCTION_ARGS)
+{
+	Oid chunk_relid = PG_GETARG_OID(0);
+	Oid chunk_table = PG_GETARG_OID(1);
+	RelationSize uncompressed_size = { .heap_size = PG_GETARG_INT64(2),
+									   .toast_size = PG_GETARG_INT64(3),
+									   .index_size = PG_GETARG_INT64(4) };
+	RelationSize compressed_size = { .heap_size = PG_GETARG_INT64(5),
+									 .toast_size = PG_GETARG_INT64(6),
+									 .index_size = PG_GETARG_INT64(7) };
+	int64 numrows_pre_compression = PG_GETARG_INT64(8);
+	int64 numrows_post_compression = PG_GETARG_INT64(9);
+	Chunk *chunk;
+	Chunk *compress_ht_chunk;
+	Cache *hcache;
+	CompressChunkCxt cxt;
+
+	Assert(!PG_ARGISNULL(0));
+	Assert(!PG_ARGISNULL(1));
+
+	TS_PREVENT_FUNC_IF_READ_ONLY();
+
+	chunk = ts_chunk_get_by_relid(chunk_relid, true);
+	hcache = ts_hypertable_cache_pin();
+	compresschunkcxt_init(&cxt, hcache, chunk->hypertable_relid, chunk_relid);
+
+	/* Acquire locks on src and compress hypertable and src chunk */
+	LockRelationOid(cxt.srcht->main_table_relid, AccessShareLock);
+	LockRelationOid(cxt.compress_ht->main_table_relid, AccessShareLock);
+	LockRelationOid(cxt.srcht_chunk->table_id, ShareLock);
+
+	/* Aquire locks on catalog tables to keep till end of txn */
+	LockRelationOid(catalog_get_table_id(ts_catalog_get(), HYPERTABLE_COMPRESSION),
+					AccessShareLock);
+	LockRelationOid(catalog_get_table_id(ts_catalog_get(), CHUNK), RowExclusiveLock);
+
+	/* Create compressed chunk DDL using existing table */
+	compress_ht_chunk = create_compress_chunk_table(cxt.compress_ht, cxt.srcht_chunk, chunk_table);
```

Comment:
So the function `create_compress_chunk_table` actually uses the existing compressed table, but creates something else? Sounds confusing :)

Reply:
I did not want to copy-paste or change compression related functions here. The only difference here is that now attached table consist compressed data, before that it was empty. If you take a look at the `compress_chunk` API